internal/entities/properties/service/service.go

```diff
@@ -0,0 +1,267 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package service provides a service to interact with properties of an entity
+package service
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/stacklok/minder/internal/db"
+	"github.com/stacklok/minder/internal/entities/properties"
+	minderv1 "github.com/stacklok/minder/pkg/api/protobuf/go/minder/v1"
+	v1 "github.com/stacklok/minder/pkg/providers/v1"
+)
+
+const (
+	// propertiesCacheTimeout is the default timeout for the cache of properties
+	propertiesCacheTimeout = 60
+	// bypassCacheTimeout is a special value to bypass the cache timeout
+	// it is not exported from the package and should only be used for testing
+	bypassCacheTimeout = -1
+)
+
+// PropertiesService is the interface for the properties service
+type PropertiesService interface {
```

Comment:
At first I wanted to have just the `Retrieve` methods that would save the entity as well when saving the properties. But - I don't know if we can generalize saving the entity, especially while we are still in migrating mode and are saving the entity with a defined UUID that we take from the older tables. So at least for now I split the methods into separate retrieve and save where the idea is that the caller would retrieve properties, save the entity and then save the properties.